app/src/main/java/net/gsantner/markor/activity/DocumentEditFragment.java

```diff
@@ -551,29 +578,25 @@ public void onFsViewerConfig(FilesystemViewerData.Options dopt) {
         }
     }
 
-    private long _lastChangedThreadStart = 0;
-
     @OnTextChanged(value = R.id.document__fragment__edit__highlighting_editor, callback = OnTextChanged.Callback.TEXT_CHANGED)
     public void onContentEditValueChanged(CharSequence text) {
-        if ((_lastChangedThreadStart + HISTORY_DELTA) < System.currentTimeMillis()) {
-            _lastChangedThreadStart = System.currentTimeMillis();
-            _hlEditor.postDelayed(() -> {
-                Activity activity = getActivity();
-                if (activity instanceof AppCompatActivity) {
-                    ((AppCompatActivity) activity).supportInvalidateOptionsMenu();
-                }
-            }, HISTORY_DELTA);
-        }
-        Activity activity = getActivity();
-        if (activity != null && activity instanceof AppCompatActivity) {
-            ((AppCompatActivity) activity).supportInvalidateOptionsMenu();
-        }
+        checkTextChangeState();
+        updateUndoRedoIconStates();
```

Comment:
As our tests and updates are fast (don't need to redraw whole menu etc). No need to use a HISTORY_DELTA delay. Can just call functions directly.

Investigating impact on performance.

Reply:
I think off adding a generic debouncer timer class to Callback.java. It's a common thing that there is a recurring pressure on some things, which inside makes even more pressure (graphics drawing).

you can keep it here without optimizations / stripping code

Reply:
In general I think it would be useful, yes. `debounce(delay, callback.b0)` or something